cylc/flow/cfgspec/workflow.py

```diff
@@ -1538,7 +1539,7 @@ def upgrade_graph_section(cfg: Dict[str, Any], descr: str) -> None:
     """Upgrade Cylc 7 `[scheduling][dependencies][X]graph` format to
     `[scheduling][graph]X`."""
     # Parsec upgrader cannot do this type of move
-    try:
+    with contextlib.suppress(KeyError):
```

Comment:
```python
with suppress(Something):
    something()
```

Is generally cleaner and more readable than:

```python
try:
    something()
except Something:
    pass
```

I'm not so fussed, if this causes us undue pain we can always disable the rule, however, it does have a potential advantage from the perspective of code coverage so thats nice.

Reply:
I quite like the `with suppress...` version.

Reply:
One thing to bear in mind is `with suppress` is slower, so we might want to avoid it sometimes
```python
from contextlib import suppress

def f():
    try:
        int('a')
    except ValueError:
        pass
    
def g():
    with suppress(ValueError):
        int('a')
```
```python
In [5]: %timeit f()
976 ns ± 46 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [6]: %timeit g()
1.63 µs ± 32.8 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```